subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNode.java

```diff
@@ -41,13 +43,15 @@
  */
 public class LocalTaskNode extends TaskNode {
     private final TaskInternal task;
+    private final WorkValidationContext validationContext;
     private ImmutableActionSet<Task> postAction = ImmutableActionSet.empty();
     private boolean isolated;
     private List<? extends ResourceLock> resourceLocks;
     private TaskProperties taskProperties;
 
     public LocalTaskNode(TaskInternal task) {
         this.task = task;
+        this.validationContext = new DefaultWorkValidationContext();
```

Comment:
We are only re-playing the type validations now because we currently skip the non-cacheable warnings in the execution plan. That I think is going to change when we will deprecate non-well-defined inputs and outputs for non-cacheable tasks, too.

I added some de-duplication logic to avoid the duplicate messages.